production/elasticsearch.yml

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright 2017 The Jaeger Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+#
+
+# This template is based on the Elasticsearch staging example: 
+# https://github.com/kubernetes/examples/tree/master/staging/elasticsearch
+# It's not intended for production purposes, but it's simple enough for 
+# an example on our Jaeger production deployment.
+
+apiVersion: v1
+kind: Template
+labels:
+  app: jaeger
+  template: jaeger-elasticsearch-template
+  jaeger-infra: template-elasticsearch
+metadata:
+  name: jaeger-elasticsearch-template
+  annotations:
+    description: Elasticsearch storage for Jaeger
+    iconClass: icon-go-gopher
+    openshift.io/display-name: Jaeger's Elasticsearch
+    tags: instant-app,tracing,opentracing,jaeger,elasticsearch
+  labels:
+    app: jaeger
+    name: jaeger-infra
+    jaeger-infra: template-production-elasticsearch
+objects:
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    name: elasticsearch
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: elasticsearch
+    labels:
+      component: elasticsearch
+  spec:
+    type: LoadBalancer
+    selector:
+      component: elasticsearch
+    ports:
+    - name: http
+      port: 9200
+      protocol: TCP
+    - name: transport
+      port: 9300
+  protocol: TCP
+- apiVersion: v1
+  kind: ReplicationController
+  metadata:
+    name: es
+    labels:
+      component: elasticsearch
+  spec:
+    replicas: 1
+    template:
+      metadata:
+        labels:
+          component: elasticsearch
+      spec:
+        serviceAccount: elasticsearch
+        containers:
+        - name: es
+          image: quay.io/pires/docker-elasticsearch-kubernetes:1.7.1-4
+          env:
+          - name: KUBERNETES_CA_CERTIFICATE_FILE
+            value: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+          - name: NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          - name: "CLUSTER_NAME"
+            value: "myesdb"
+          - name: "DISCOVERY_SERVICE"
+            value: "elasticsearch"
+          - name: NODE_MASTER
+            value: "true"
+          - name: NODE_DATA
+            value: "true"
+          - name: HTTP_ENABLE
+            value: "true"
+          ports:
+          - containerPort: 9200
+            name: http
+            protocol: TCP
+          - containerPort: 9300
+            name: transport
+            protocol: TCP
+          volumeMounts:
```

Comment:
does this mount volume from the host?

Reply:
No: it mounts at `/data` the storage named `storage`, of type `emptyDir`. So, it's not persistent across reboots.

Reply:
thanks